tensorflow/dtensor/mlir/ir/tf_dtensor.td

```diff
@@ -664,4 +664,28 @@ sharded than the input layout.
   let hasVerifier = 1;
 }
 
+def TF_DTensorAllToAllOp : TF_Op<"DTensorAllToAll", [Pure]> {
+  let summary = "Mutually exchanges the input to match the given layout.";
+
+  let description = [{
+This op takes both an input and an output layout. There can be one mesh
+dimension which is becoming unsharded in one axis while becoming sharded 
+in another axis."
+  }];
+
+  let arguments = (ins
+    TensorOf<[TF_Bfloat16, TF_Float16, TF_Float32, TF_Int32, TF_Uint32, TF_Int64, TF_Bool]>:$input,
+    DTensor_LayoutAttr:$input_layout,
```

Comment:
Nonblocking: An Op that takes group_assignment and dimensions as input may make fusing multiple AllToAll into one easier. For example, DTensorAllReduce, which has a fusing optimization, takes a group_assignment input. The reason I say that is because multiple layouts can be reduced to the same group_assignment, and most likely all-and-only ops using the same group_assignment can be fused into one. We can worry about this in more details later.

Reply:
Got it, we can definitely change this later.